packages/react/src/primitives/StepperField/useStepper.ts

```diff
@@ -50,11 +50,11 @@ export const useStepper = ({
 
   // Same for controlled components on the first render because users could provide invalid intial value.
   // It seems redundant afterwards but necessary for the first render
-  if (isControlled) {
+  if (controlledValue) {
     controlledValue = getCorrectSteppingValue(min, max, step, controlledValue);
   }
 
-  const value = isControlled ? controlledValue : uncontrolledValue;
+  const value = controlledValue ? controlledValue : uncontrolledValue;
```

Comment:
To prevent the mutation of `controlledValue` on line 53, we can just call `getCorrectSteppingValue` (kudos on that name to whoever came up with it, maybe @zchenwei ?) in this ternary:
```
const value = controlledValue
    ? getCorrectSteppingValue(controlledValue)
    : uncontrolledValue;
```